knimin/static/js/platemapper.js

```diff
@@ -0,0 +1,506 @@
+        if ( item.category ) {
+          li.attr( "aria-label", item.category + " : " + item.label );
+        }
+        });
+      }
+    });
+});
+
+/**
+ *
+ * @class PlateMap
+ *
+ * Represents a Plate
+ *
+ * @param {string} target The name of the target container for the plate map
+ * @param {int} rows The number of rows in the plate
+ * @param {int} cols The number of columns in the plate
+ * @param {string[]} samples The list of samples to populate the autocomplete
+ *
+ * @return {PlateMap}
+ * @constructs PlateMap
+ *
+ **/
+function PlateMap(target, plate_id) {
+  var obj = this;
+
+  // Store the plate ID and the target DOM element
+  this.plateId = plate_id
+  this.target = $('#' + target)
+
+  // Retrieve the plate information from the server
+  $.get('/pm_sample_plate?plate_id=' + plate_id, function (data) {
+    obj.initialize(data);
+  })
+    .fail(function (jqXHR, textStatus, errorThrown) {
+      if(jqXHR.status === 404) {
+        var data = $.parseJSON(jqXHR.responseText)
+        $('<h3 class="warning">').html(data.message).appendTo(obj.target);
+      } else {
+        $('<div>').html(jqXHR.responseText).appendTo(obj.target);
+      }
+    });
+};
+
+/**
+ *
+ * Initializes the object after the GET query is completed
+ *
+ * @param {Object} data The data returned from the GET query
+ *
+ **/
+PlateMap.prototype.initialize = function (data) {
+    var study, color;
+
+    this.name = data.name;
+    this.createdOn = data.created_on;
+    this.createdBy = data.email;
+    this.notes = data.notes;
+    this.plateType = data.plate_type.notes;
+    this.rows = data.plate_type.rows;
+    this.cols = data.plate_type.cols;
+    this.studies = data.studies;
+
+    // Construct a dictionary keyed by sample, for easy access to the sample
+    // information
+    this.samples = {}
+    // This is a special list needed for initializing the sample autocompletion
+    this.autoCompleteSamples = []
+    // Iterate over all the studies
+    for (var idx = 0; idx < this.studies.length; idx++) {
+      study = this.studies[idx];
+      color = PlateMap._qiimeColors[idx];
+      // Iterate over all samples
+      for (var sample of study.samples.all) {
+        this.samples[sample] = {color: color, plates: []};
+        this.autoCompleteSamples.push({label: sample, category: study.title, color: color});
+      }
+      // Iterate over all plates to get the already plated samples
+      for (var plate in study.samples.plated) {
+        if (study.samples.plated.hasOwnProperty(plate)) {
+          this.samples[sample].plates.push(plate);
+        }
+      }
+    }
+
+    // Create a 2D array to store the per well information
+    this.wellInformation = new Array(this.rows);
+    for (var i = 0; i < this.wellInformation.length; i++) {
+      this.wellInformation[i] = new Array(this.cols);
+      for (var j = 0; j < this.wellInformation[i].length; j++) {
+        this.wellInformation[i][j] = {inputTag: undefined, comment: undefined};
```

Comment:
I would use null instead of undefined, as the value of undefined could be true for any other non-existing property i.e. if you called `this.wellInformation[i][j].yoshiki` that value would also be `undefined`, `null` would be a different value *at least*.